The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/3234203462/jobs/5297020102) [(plain)](https://github.com/rust-lang/rust/commit/3aad3cdc41d9a2421feab1d94470b2db28b226f5/checks/8844433564/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
configure: rust.debug-assertions := True
configure: rust.overflow-checks := True
configure: llvm.assertions      := True
configure: dist.missing-tools   := True
configure: build.configure-args := ['--enable-sccache', '--disable-manage-submodu ...
configure: writing `config.toml` in current directory
configure: 
configure: run `python /checkout/x.py --help`
Attempting with retry: make prepare
---
---- src/ffi/c_str.rs - ffi::c_str::CStr::from_ptr (line 236) stdout ----
error[E0133]: call to unsafe function is unsafe and requires unsafe function or block
  --> src/ffi/c_str.rs:245:22
   |
12 | const HELLO: &CStr = CStr::from_ptr(HELLO_PTR);
   |
   = note: consult the function's documentation for information on how to avoid undefined behavior

error: aborting due to previous error
---
    src/ffi/c_str.rs - ffi::c_str::CStr::from_ptr (line 236)

test result: FAILED. 3913 passed; 1 failed; 36 ignored; 0 measured; 0 filtered out; finished in 45.64s

error: doctest failed, to rerun pass `-p core --doc`
```

</details>

--- a/library/core/src/ffi/c_str.rs
+++ b/library/core/src/ffi/c_str.rs
@@ -221,9 +221,7 @@ impl CStr {
     /// # Examples
     ///
     /// ```ignore (extern-declaration)
-    /// # fn main() {
-    /// use std::ffi::CStr;
-    /// use std::os::raw::c_char;
+    /// use std::ffi::{c_char, CStr};
     ///
     /// extern "C" {
     ///     fn my_string() -> *const c_char;
@@ -233,7 +231,18 @@ impl CStr {
     ///     let slice = CStr::from_ptr(my_string());
     ///     println!("string returned: {}", slice.to_str().unwrap());
     /// }
-    /// # }
+    /// ```
+    ///
+    /// ```
+    /// #![feature(const_cstr_methods)]
+    ///
+    /// use std::ffi::{c_char, CStr};
+    ///
+    /// const HELLO_PTR: *const c_char = {
+    ///     const BYTES: &[u8] = b"Hello, world!\0";
+    ///     BYTES.as_ptr().cast()
+    /// };
+    /// const HELLO: &CStr = unsafe { CStr::from_ptr(HELLO_PTR) };
     /// ```
     ///
     /// [valid]: core::ptr#safety

